File: code/mathematical_algorithms/src/greatest_digit_in_number/greatest_digit_in_number.java
/* Part of Cosmos by OpenGenus Foundation */
import java.util.*;
public class Greatest_digit_in_number{
    public static void main(String args[]) {
        int num = 23238934;
        System.out.println(greatest_digit(num));
    }
    
    // returns greatest digit in a given number
    public static int greatest_digit(int n)
    {
        int max = 0;
        while(n != 0)
        {
            if(max<n%10) max=n%10;
            n/=10;
        }
        return max;
    }
}

File: code/mathematical_algorithms/src/greatest_digit_in_number/greatest_digit_in_number.js
/*

// mathematical algorithms | greatest digit in number | greatest digit in number | JAVASCRIPT
// Part of Cosmos by OpenGenus Foundation

*/


function greatestDigit(num) {
  return String(num)
    .split("")
    .reduce((a, b) => Math.max(a, b));
}

console.log(greatestDigit(123)); // 3
console.log(greatestDigit(321)); // 3
console.log(greatestDigit(12903)); // 9
